rust/agama-cli/src/questions.rs

```diff
@@ -35,30 +39,54 @@ pub enum Modes {
     NonInteractive,
 }
 
-async fn set_mode(proxy: Questions1Proxy<'_>, value: Modes) -> anyhow::Result<()> {
-    // TODO: how to print dbus error in that anyhow?
+async fn set_mode(proxy: Questions1Proxy<'_>, value: Modes) -> Result<(), ServiceError> {
     proxy
         .set_interactive(value == Modes::Interactive)
         .await
-        .context("Failed to set mode for answering questions.")
+        .map_err(|e| e.into())
 }
 
-async fn set_answers(proxy: Questions1Proxy<'_>, path: String) -> anyhow::Result<()> {
-    // TODO: how to print dbus error in that anyhow?
+async fn set_answers(proxy: Questions1Proxy<'_>, path: String) -> Result<(), ServiceError> {
     proxy
         .add_answer_file(path.as_str())
         .await
-        .context("Failed to set answers from answers file")
+        .map_err(|e| e.into())
 }
 
-pub async fn run(subcommand: QuestionsCommands) -> anyhow::Result<()> {
+async fn list_questions() -> Result<(), ServiceError> {
+    let client = HTTPClient::new().await?;
+    let questions = client.list_questions().await?;
+    // FIXME: that conversion to anyhow error is nasty, but we do not expect issue
```

Comment:
We could have a generic error for these cases. But let's keep it as it is now. At some point we should improve our error types.